src/gateways/ElasticsearchGateway.ts

```diff
@@ -31,7 +27,30 @@ export class ElasticsearchGateway {
     await this.client.index({
       index: this.indexName,
       id: metadata.documentId,
-      body: metadata
+      body: metadata,
     });
   }
+
+  async findDocuments(metadata: DocumentMetadata): Promise<string[]> {
+    this.logger
+      .mergeContext({ indexName: this.indexName })
+      .log('[elasticsearch] searching documents');
+
+    const conditionsArray = Object.entries(metadata).map((condition) => {
```

Comment:
What does a condition look like? I've had a look at the DocumentMetadata interface but that doesn't tell me a lot. Does the interface need updating?

Reply:
So I think metadata is something like:
`
{
name: 'jess',
dob:'yesterday'
}
`

Then that turned into an array would be:
`
[ [ 'name', 'jess' ], [ 'dob', 'yesterday' ] ]
`

Then it gets mapped into a format that can be used for es searching, so in this case:
`conditionsArray = [ { match: { name: 'jess' } }, { match: { dob: 'yesterday' } } ]`

And that gets used on line 48

I don't think it has to do with the interface, I think it's just the format that es request requires

Reply:
OK I see, the naming of "condition" threw me a little in the function. You can remove the Object.fromEntries by 

```Object.entries({ name: 'jess', dob:'yesterday' }).map(([key, value]) => ({match: {[key]: value}}));```